simulariumio/trajectory_converter.py

```diff
@@ -358,6 +361,43 @@ def _check_agent_ids_are_unique_per_frame(buffer_data: Dict[str, Any]) -> bool:
                 get_n_subpoints = True
         return True
 
+    @staticmethod
+    def _check_types_match_subpoints(trajectory_data: TrajectoryData) -> str:
```

Comment:
just curious, I wonder if this could be done per-agent rather than iterating over all the data.